bip-tx-terminology.mediawiki

```diff
@@ -0,0 +1,286 @@
+*** Second '''Witness Stack''': The Witness data corresponding to the second input.
+**** Item count for the second '''Witness Stack''' <code>03</code>: serialization artifact, two or more Witness Items indicate a scriptpath spend.
+**** First '''Witness Item'''
+***** '''Length of the first Witness Item''' <code>40</code>: 64 bytes indicate a signature
+***** '''Signature''' <code>7b5d614a4610bf9196775791fcc589597ca066dcd10048e004cd4c7341bb4bb90cee4705192f3f7db524e8067a5222c7f09baf29ef6b805b8327ecd1e5ab83ca</code>: script argument used as an input to the condition script
+**** '''Second Witness Item'''
+***** Length of the second '''Witness Item''' <code>22</code>: serialization artifact instructing to read 34 bytes
+***** '''Leaf Script''' <code>20f5b059b9a72298ccbefff59d9b943f7e0fc91d8a3b944a95e7b6390cc99eb5f4ac</code>: the condition scripts for the P2TR scriptpath spend
+**** Third '''Witness Item'''
+***** Length of the third '''Witness Item''' <code>41</code>: serialization artifact, the length indicates the depth of the Control Block and how many hashing partners are necessary to prove the membership of the Leaf Script
+***** '''Leaf Version''' <code>c0</code>: indicates the version of Tapscript used in the Leaf Script
+***** '''Inner key''' <code>d9dfdf0fe3c83e9870095d67fff59a8056dad28c6dfb944bb71cf64b90ace9a7</code>: The Taproot Internal Key which was tweaked with the Merkle root of the Script Tree to result in the Public Key committed to in the UTXO's Witness Program being spent
+***** '''Tap Branch''' <code>776b22a1185fb2dc9524f6b178e2693189bf01655d7f38f043923668dc5af45b</code>: The hashing partners necessary to connect the Leaf Script to the Merkle root of the Script Tree.
+** '''Transaction Header (cont)''': A collective term to refer to the meta information of a transaction, split across the transaction.
+*** '''Locktime''' <code>ffd30a00</code>: the 4-byte locktime field, little endian for 709631
+
+==Rationale==
+
+; Amount vs Value
+: Amount underscores that we are referring to a count of satoshis rather than subjective value. As Bitcoin transactions (and code altogether) are littered with “values”, we prefer “amount” as being less ambiguous.
+
+; Witness
+: Ambiguous, use Witness Stack to refer to the witness data of a single Transaction Input, use Witness Structure when talking about all witness data of a transaction.
+
+; Bare Output Scripts
+: Include P2PK, P2PKH, P2MS, and arbitrary other condition scripts directly defined in the Output Script.
+
+; Flag
+: While the Flag currently only indicates the presence of a Witness Structure, it could perhaps also indicate the presence of other extended serialization formats in the future, therefore, it’s only flag, not “witness flag”.
+
+; P2SH Program
```

Comment:
Agree very much with this. It felt like it was a missing term so I'm glad to see it has been given a name!